### PR TITLE
fix: use correct TF version in empty framework_version warning

### PR DESCRIPTION
**Pull request by laurenyu**
*Description of changes:*
Addresses #754 (`LATEST_VERSION` wasn't being used in the TensorFlow estimator)

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#commit-message-guidlines)
- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have updated any necessary [documentation](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: FAILED
* [Build Logs](https://9dzx6wxcy9.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=7d951217-ae2b-44a1-95d0-3046a8bb8b97%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: src/sagemaker/tensorflow/estimator.py [hunk 1]
```diff
@@ -229,7 +229,7 @@ def __init__(self, training_steps=None, evaluation_steps=None, checkpoint_path=N
             **kwargs: Additional kwargs passed to the Framework constructor.
         """
         if framework_version is None:
-            LOGGER.warning(fw.empty_framework_version_warning(TF_VERSION, TF_VERSION))
+            LOGGER.warning(fw.empty_framework_version_warning(TF_VERSION, self.LATEST_VERSION))
```
Comment (Nishanksingla) [inline src/sagemaker/tensorflow/estimator.py]: self.LATEST_VERSION looks wrong to me as LATEST_VERSION is not class variable, it is defined outside the __init__. 
It should be just LATEST_VERSION

Comment (laurenyu) [reply src/sagemaker/tensorflow/estimator.py]: it's defined in the class `TensorFlow`: https://github.com/aws/sagemaker-python-sdk/blob/42c7f308e417d945680e200d3e80743e43ab50d3/src/sagemaker/tensorflow/estimator.py#L170-L174

`LATEST_VERSION` (without `self`) would work only if it is defined outside of the `TensorFlow` class:
```python
>>> class A:
...     FOO = 1
...     def __init__(self):
...         print(FOO)
... 
>>> A()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 4, in __init__
NameError: name 'FOO' is not defined
```

Comment (Nishanksingla) [reply src/sagemaker/tensorflow/estimator.py]: Ok Thanks for the explanation.

File: src/sagemaker/tensorflow/estimator.py [hunk 2]
```diff
@@ -229,7 +229,7 @@ def __init__(self, training_steps=None, evaluation_steps=None, checkpoint_path=N
...
         self.framework_version = framework_version or TF_VERSION
```
Comment (Nishanksingla) [inline src/sagemaker/tensorflow/estimator.py]: Still you haven't changed TF_VERSION to LATEST_VERSION

Comment (Nishanksingla) [reply src/sagemaker/tensorflow/estimator.py]: The initial reported issue is that TF_VERSION is set '1.11' and LATEST_VERSION is set '1.12'. TF_VERSION should be replaced by LATEST_VERSION. Or TF_VERSION should be updated to 1.12

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: FAILED
* [Build Logs](https://9dzx6wxcy9.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=27c84927-0190-4108-a793-128b5d0b2888%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: src/sagemaker/tensorflow/estimator.py (hunk 2 above)
Comment (laurenyu) [reply src/sagemaker/tensorflow/estimator.py]: I understand that the transition from having an updating default value to requiring `framework_version` be explicitly defined could be better, and I apologize for the inconvenience. There's more explanation in #754 around this, and there'll be a separate PR for the documentation changes proposed in #754.

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: FAILED
* [Build Logs](https://9dzx6wxcy9.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=3a993b39-f931-4ffe-8601-a5530a47b6f2%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: FAILED
* [Build Logs](https://9dzx6wxcy9.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=ed1ee34b-5770-4944-b927-8754add33ddd%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://9dzx6wxcy9.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=1537b7cc-f060-4de7-92d6-ed220ecbb7a0%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by laurenyu** [comment]
will make a separate PR to address the doc concerns raised in #754

**Pull request merged**